### ISSUE TITLE
Use juju commands instead of yaml files?

### ISSUE THREAD

**Issue by tvansteenburgh** [juju/python-libjuju#32]
When using connection shortcuts like `connect_current()` and `connect_model()`, the information necessary to make the connection is obtained from yaml files in ~/.local/share/juju/.

I just saw a case where a model was present (showed in the output of `juju list-models`), yet did not exist in ~/.local/share/juju/models.yaml yet.

To avoid this we should use the output of juju commands instead of using the yaml files directly. Either way assumes the juju cli is installed. To connect without the juju cli installed, the plain `connect()` method can be used.

**Comment by sfeole** [comment]
The only time I've seen a problem is when creating a new model via the api,  

After creating the model,   juju list-models  via the CLI did not reflect that the model existed only until I deployed something to the newly created model.  But I think that's by design. 

 I believe the act of creating a model does write to ~/.local/share/juju/model.yaml  right?   Also  I noticed when the model is deleted via the API, it does not remove itself from ~/.local/share/juju/model.yaml file. 

This is not the standard case when a model is created/destroyed via the juju commands
